### ISSUE TITLE
Unknown crash during run

### ISSUE THREAD

**Issue by smaccoun** [oven-sh/bun#10871]
### How can we reproduce the crash?

I have no idea. Keeps happening though. I will see if I can find the part of code soon where it's happening. Some important info

1. It's happening while running backend server
2. Seems to have just started on 1.1.7

### JavaScript/TypeScript code that reproduces the crash?

Not sure yet. Will try to figure out. Happening on backend using lots of trpc, kysely, aws sdk

### Relevant log output

```shell
Bun v1.1.7 (b0b7db5c) macOS Silicon
Args: "bun", "--hot", "index.ts"
Features: jsc abort_signal dotenv fetch(10) http_server transpiler_cache(102) tsconfig_paths(3) tsconfig(23)
Builtins: "abort-controller" "bun:main" "node:assert" "node:async_hooks" "node:buffer" "node:child_process" "node:constants" "node:crypto" "node:dns" "node:events" "node:fs" "node:http" "node:https" "node:module" "node:net" "node:os" "node:path" "node:querystring" "node:readline" "node:stream" "node:stream/promises" "node:string_decoder" "node:tls" "node:tty" "node:url" "node:util" "node:util/types" "node:zlib" "node:punycode" "ws" "node-fetch" "utf-8-validate" "node:http2"
Elapsed: 608009ms | User: 3157ms | Sys: 1120ms
RSS: 0.57GB | Peak: 0.57GB | Commit: 65.24MB | Faults: 144
```

### Stack Trace (bun.report)

Bun v1.1.7 ([`b0b7db5`](https://github.com/oven-sh/bun/tree/b0b7db5c060e830801b0273b7ff6e34e3eb5c254)) on macos aarch64 [AutoCommand]

Segmentation fault at address 0x00000008

- *2 unknown/js code*
- `JSC::InlineCacheCompiler::regenerate`
- `JSC::StructureStubInfo::addAccessCase`
- `operationInstanceOfOptimize`
- *5 unknown/js code*

**Comment by Jarred-Sumner** [comment]
Judging by the stack trace, this looks like https://github.com/WebKit/WebKit/pull/28106 might fix it, but we haven't merged that PR yet

**Comment by Jarred-Sumner** [comment]
This should be fixed as of Bun v1.1.8. Please let us know if you're still running into this
